sql/core/src/test/java/test/org/apache/spark/sql/JavaSaveLoadSuite.java

```diff
@@ -43,7 +43,7 @@
   Dataset<Row> df;
 
   private static void checkAnswer(Dataset<Row> actual, List<Row> expected) {
-    String errorMessage = QueryTest$.MODULE$.checkAnswer(actual, expected);
+    String errorMessage = QueryTest$.MODULE$.getErrorMessageInCheckAnswer(actual, expected);
     if (errorMessage != null) {
```

Comment:
This construction seems kinda pointless then. Why not just let checkAnswer fail with an assertion, rather than have a separate method that only returns the error and have tests manually check if there's an error string and fail it?

Reply:
If there's no issue on calling `fail` from scalatest in java suite (mixed-up usage), that seems the way to go; otherwise we may be able to have another QueryTest as utility class for java suite, and move this method instead and co-use across java suites.

Reply:
I think it's OK; it will generate a relevant failure exception and message. I'd prefer it for simplicity.